cpp/src/arrow/array-list-test.cc

```diff
@@ -332,4 +332,219 @@ TEST_F(TestListArray, TestBuilderPreserveFieleName) {
+  ASSERT_OK(builder_->AppendNull());
+  ASSERT_OK(builder_->AppendNull());
+
+  Done();
+
+  ASSERT_OK(ValidateArray(*result_));
+  ASSERT_TRUE(result_->IsNull(0));
+  ASSERT_TRUE(result_->IsNull(1));
+
+  ASSERT_EQ(0, result_->value_offset(0));
+  ASSERT_EQ(list_size(), result_->value_offset(1));
+
+  auto values = result_->values();
+  ASSERT_EQ(list_size() * 2, values->length());
+}
+
+TEST_F(TestFixedSizeListArray, TestAppendNulls) {
+  ASSERT_OK(builder_->AppendNulls(3));
+
+  Done();
+
+  ASSERT_OK(ValidateArray(*result_));
+  ASSERT_EQ(result_->length(), 3);
+  ASSERT_EQ(result_->null_count(), 3);
+  ASSERT_TRUE(result_->IsNull(0));
+  ASSERT_TRUE(result_->IsNull(1));
+  ASSERT_TRUE(result_->IsNull(2));
+
+  ASSERT_EQ(0, result_->value_offset(0));
+  ASSERT_EQ(list_size(), result_->value_offset(1));
+  ASSERT_EQ(list_size() * 2, result_->value_offset(2));
+
+  auto values = result_->values();
+  ASSERT_EQ(list_size() * 3, values->length());
+}
+
+void ValidateBasicFixedSizeListArray(const FixedSizeListArray* result,
+                                     const std::vector<int32_t>& values,
+                                     const std::vector<uint8_t>& is_valid) {
+  ASSERT_OK(ValidateArray(*result));
+  ASSERT_EQ(1, result->null_count());
+  ASSERT_LE(result->values()->null_count(), 2);
+
+  ASSERT_EQ(3, result->length());
+  for (int32_t i = 0; i < 3; ++i) {
+    ASSERT_EQ(i * result->value_length(), result->value_offset(i));
+  }
+
+  for (int i = 0; i < result->length(); ++i) {
+    ASSERT_EQ(is_valid[i] == 0, result->IsNull(i));
+  }
+
+  ASSERT_EQ(result->length() * result->value_length(), result->values()->length());
+  auto varr = std::dynamic_pointer_cast<Int32Array>(result->values());
+
+  for (size_t i = 0; i < values.size(); ++i) {
+    if (is_valid[i / result->value_length()] == 0) {
+      continue;
+    }
+    ASSERT_EQ(values[i], varr->Value(i));
+  }
+}
+
+TEST_F(TestFixedSizeListArray, TestBasics) {
+  std::vector<int32_t> values = {0, 1, 2, 3, 4, 5};
+  std::vector<uint8_t> is_valid = {1, 0, 1};
+
+  Int32Builder* vb = checked_cast<Int32Builder*>(builder_->value_builder());
+
+  ASSERT_OK(builder_->Reserve(values.size() / list_size()));
+  ASSERT_OK(vb->Reserve(values.size()));
+
+  int pos = 0;
+  for (size_t i = 0; i < values.size() / list_size(); ++i) {
+    if (is_valid[i] == 0) {
+      ASSERT_OK(builder_->AppendNull());
+      pos += list_size();
+      continue;
+    }
+    ASSERT_OK(builder_->Append());
+    for (int j = 0; j < list_size(); ++j) {
+      ASSERT_OK(vb->Append(values[pos++]));
+    }
+  }
+
+  Done();
+  ValidateBasicFixedSizeListArray(result_.get(), values, is_valid);
+}
+
+TEST_F(TestFixedSizeListArray, BulkAppend) {
+  std::vector<int32_t> values = {0, 1, 2, 3, 4, 5};
+  std::vector<uint8_t> is_valid = {1, 0, 1};
+
+  Int32Builder* vb = checked_cast<Int32Builder*>(builder_->value_builder());
+
+  ASSERT_OK(builder_->Reserve(values.size() / list_size()));
```

Comment:
Is it required to `Reserve` explicitly here? The builder should reserve automatically when the safe append methods are called.

Reply:
You're correct. It's just copypasta from the List test cases; I'll remove it.